fabtests/unit/mr_test.c

```diff
@@ -188,16 +206,73 @@ static int mr_regattr()
 	return TEST_RET_VAL(ret, testret);
 }
 
+static int mr_reg_free_then_alloc()
+{
+	int i, ret, testret;
+	size_t buf_size = test_size[test_cnt-1].size;
+	struct fid_mr *mr;
+	char *buf2;
+	struct iovec iov;
+	struct fi_mr_attr attr = {0};
+
+	iov.iov_base = buf;
+	iov.iov_len = buf_size;
+
+	attr.mr_iov = &iov;
+	attr.iov_count = 1;
+	attr.access = FI_SEND;
+	attr.offset = 0;
+	attr.requested_key = FT_MR_KEY;
+	attr.context = NULL;
+	attr.iface = opts.iface;
+	if (opts.iface == FI_HMEM_CUDA)
+		attr.device.cuda = opts.device;
+
+	testret = FAIL;
+	for (i=0; i < 5; ++i) {
+		ret = fi_mr_regattr(domain, &attr, 0, &mr);
+		if (ret) {
+			FT_UNIT_STRERR(err_buf, "fi_mr_reg failed", ret);
+			goto out;
+		}
+
+		ret = fi_close(&mr->fid);
+		if (ret) {
+			FT_UNIT_STRERR(err_buf, "fi_close failed", ret);
+			goto out;
+		}
+
+		ret = ft_hmem_alloc(opts.iface, opts.device, (void **)&buf2, buf_size);
```

Comment:
I guess we allocate a buf2 and make buf = buf2 is to make sure the new alloc has different region with the existing one. Right?

Reply:
yes